File: Baekjoon/2021 KAIST RUN Spring Open Contest/A번-Bingo.cpp
#include <bits/stdc++.h>
using namespace std;
#define FOR(i,m,n) for(int i=(m);i<(n);i++)
#define REP(i,n) FOR(i,0,n)
#define ALL(v) (v).begin(),(v).end()
using ll = long long;
constexpr int INF = 0x3f3f3f3f;
constexpr long long LINF = 0x3f3f3f3f3f3f3f3fLL;
constexpr double EPS = 1e-8;
constexpr int MOD = 1000000007;
// constexpr int MOD = 998244353;
constexpr int dy[] = {1, 0, -1, 0}, dx[] = {0, -1, 0, 1};
constexpr int dy8[] = {1, 1, 0, -1, -1, -1, 0, 1}, dx8[] = {0, -1, -1, -1, 0, 1, 1, 1};

int n,k;
int q[1000];

   
    
int main(){
    ios::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);

    

    cin >> n >> k;
    int f = n - 2;
    REP(p,n){
        if(p == 0){
            q[p] = p;
        }
        else if(p == n - 1){
            q[p] = p;
        }
        else{
            q[p] = f;
            f--;
        }
    }

    if(n == 2){
        if(k > 1){
            cout << "NO";
            exit(0);
        }
        
    }
   
    if(k <= n*n -n){
        cout << "YES" << "\n";
        int cnt = 0;

        REP(i,n){
            REP(j,n){
                if(j == q[i]){
                    cout << ".";
                    continue;
                }
                if( cnt < k ){
                    cout << "#";
                    cnt++;
                }
                else cout << ".";

            }
            cout << "\n";
        }
        }
    else cout << "NO";


}

File: Baekjoon/2021 KAIST RUN Spring Open Contest/A번-Bingo.py
import sys
INF = sys.maxsize
input = sys.stdin.readline
sys.setrecursionlimit(10**6)
MOD = 1000000007
dy = [1, 0, -1, 0]
dx = [0, -1, 0, 1]
dy8 = [1, 1, 0, -1, -1, -1, 0, 1]
dx8 = [0, -1, -1, -1, 0, 1, 1, 1]

def li(): return list(map(int,input().split()))

n,k = li()
s = n -2
q = [ 0 for _ in range(n)]
for i in range(n):
    if i == 0:
        q[i] = i
    elif i == n - 1:
        q[i] = i
    else:
        q[i] = s
        s -= 1

if n == 2:
    if k > 1: 
        print("NO")
        quit()

if k <= n*n -n:
        print("YES")
        cnt = 0

        for i in range(n):
            for j in range(n):
                if j == q[i]:
                    print("." , end = "")
                    continue
                if cnt < k :
                    print("#", end="")
                    cnt += 1
                else: print(".", end= "")
            print()
        
else: print("NO")
    

